src/mds/MDSRank.cc

```diff
@@ -1740,6 +1737,61 @@ void MDSRankDispatcher::handle_mds_map(
   if (g_conf->mds_dump_cache_on_map)
     mdcache->dump_cache();
 
+  // mdsmap and oldmap can be discontinuous. failover might happen in the missing mdsmap.
+  set<mds_rank_t> restart;
+  if (state >= MDSMap::STATE_RESOLVE) {
```

Comment:
I assume we don't need to mark the down MDSs when `< MDSMap::STATE_RESOLVE` because the MDS won't have any connections yet with the other ranks? Can you please add a comment?